src/settings.js

```diff
@@ -1419,3 +1419,17 @@ var TARGET_BASENAME = '';
 // this to 0 to save a little bit of code size and performance when catching exceptions.
 var SUPPORT_LONGJMP = 1;
 
+// Legacy settings that have been removed, and the values they are now fixed to. These can no
+// longer be changed:
+// [OPTION_NAME, POSSIBLE_VALUES, ERROR_EXPLANATION], where POSSIBLE_VALUES is an array of values that will
+// still be silently accepted by the compiler. First element in the list is the canonical/fixed value going forward.
+// This allows existing build systems to keep specifying one of the supported settings, for backwards compatibility.
+var REMOVED_SETTINGS = [
```

Comment:
Can we move this directly into the shared.py python code?  There is no need to for REMOVED_SETTINGS to be part of settings.js is there?

Reply:
Having these in src/settings.js is intentional for two reasons. First one is that people currently know to look in src/settings.js for options, and they rely on the documentation in there. Having these in src/settings.js helps people discover them to figure out which settings have gone away. Also, having the options in src/settings.js helps the process of removing options (just move them in the array at the end of the same file).

If the info related to deprecated options was in a separate file than the options themselves, users will not discover them easy, and it can be easier to forget to update the list when removing options.

Reply:
But the comment right above already says "Legacy settings that have been removed...".

I would argue that this CL effectively means these settings are still supported (albeit with a fixed value).  IIUC, they will also still be supported in user code too.   User JS code can still check `SPLIT_MEMORY` right?   They are settings that we are carrying over in some form from an earlier time, to me this what legacy means.

Reply:
This is getting to odd name bikeshedding. The options were first legacy, and then the options have now been removed. I don't think it makes sense to argue over "is a setting a setting if its value is now fixed". I'll just change the name.